test/powershell/Language/Operators/PipelineChainOperator.Tests.ps1

```diff
@@ -0,0 +1,338 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe "Experimental Feature: && and || operators - Feature-Enabled" -Tag CI {
+    BeforeAll {
+        $skipTest = -not $EnabledExperimentalFeatures.Contains('PSPipelineChainOperators')
+        if ($skipTest)
+        {
+            Write-Verbose "Test Suite Skipped: These tests require the PSPipelineChainOperators experimental feature to be enabled" -Verbose
+            $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+            $PSDefaultParameterValues["it:skip"] = $true
+            return
+        }
+
+        function Test-SuccessfulCommand
+        {
+            Write-Output "SUCCESS"
+        }
+
+        filter Test-NonTerminatingError
+        {
+            [CmdletBinding()]
+            param(
+                [Parameter(ValueFromPipeline)]
+                [object[]]
+                $Input
+            )
+
+            if ($Input -ne 2)
+            {
+                return $Input
+            }
+
+            $exception = [System.Exception]::new("NTERROR")
+            $errorId = "NTERROR"
+            $errorCategory = [System.Management.Automation.ErrorCategory]::NotSpecified
+
+            $errorRecord = [System.Management.Automation.ErrorRecord]::new($exception, $errorId, $errorCategory, $null)
+            $PSCmdlet.WriteError($errorRecord)
+        }
+
+        $simpleTestCases = @(
+            # Two native commands
+            @{ Statement = "testexe -returncode -1 && testexe -echoargs 'A'"; Output = @('-1') }
+            @{ Statement = "testexe -returncode -1 || testexe -echoargs 'A'"; Output = '-1','Arg 0 is <A>' }
+            @{ Statement = "testexe -returncode 0 && testexe -echoargs 'A'"; Output = '0','Arg 0 is <A>' }
+            @{ Statement = "testexe -returncode 0 || testexe -echoargs 'A'"; Output = @('0') }
+            @{ Statement = 'testexe -returncode 0 > $null && testexe -returncode 1'; Output = @('1') }
+            @{ Statement = 'testexe -returncode 0 && testexe -echoargs ''A'' > $null'; Output = @('0') }
```

Comment:
NIT: this line and the previous line use single quotes whereas the previous ones use double quotes, this also causes A in this line to use two single quotes.  Best would be to use single quotes for all of these and double quotes around A.